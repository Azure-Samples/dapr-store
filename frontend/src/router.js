import VueRouter from 'vue-router'

import ProductCatalog from './components/ProductCatalog'
import ProductOffers from './components/ProductOffers'
import Home from './components/Home'

const router = new VueRouter({
  routes: [
    {
      path: '/catalog',
      component: ProductCatalog
    },
    {
      path: '/offers',
      component: ProductOffers
    },
    {
      path: '/',
      component: Home
    }
  ]
})

export default router