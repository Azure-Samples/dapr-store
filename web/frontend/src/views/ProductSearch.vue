<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Show product results from a search
// ----------------------------------------------------------------------------
-->

<template>
  <div>
    <h1>Search results for: <i>{{ this.$route.params.query }}</i></h1>
    <error-box :error="error" />
    <product-list v-if="!error" :products="products" />
  </div>
</template>

<script>
import api from '../services/api'
import ProductList from '../components/ProductList'
import ErrorBox from '../components/ErrorBox'

export default {
  name: 'ProductSearch',

  components: {
    'product-list': ProductList,
    'error-box': ErrorBox
  },

  data() {
    return {
      products: null,
      error: null
    }
  },

  watch: {
    async $route() {
      this.doSearch()
    }
  },

  async mounted() {
    this.doSearch()
  },

  methods: {
    async doSearch() {
      try {
        this.products = null
        this.products = await api.productSearch(this.$route.params.query)
      } catch (err) {
        this.error = err
      }
    }
  }
}
</script>
