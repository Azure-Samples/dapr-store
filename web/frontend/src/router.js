// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - client side routes
// ----------------------------------------------------------------------------

import { createRouter, createWebHistory } from 'vue-router'

import auth from './services/auth'

import ProductCatalog from './views/ProductCatalog.vue'
import ProductOffers from './views/ProductOffers.vue'
import ProductSearch from './views/ProductSearch.vue'
import ProductSingle from './views/ProductSingle.vue'
import ViewOrder from './views/ViewOrder.vue'
import Login from './views/Login.vue'
import Account from './views/Account.vue'
import Home from './views/Home.vue'
import About from './views/About.vue'
import Cart from './views/Cart.vue'

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      name: 'home',
      path: '/',
      component: Home
    },
    {
      name: 'login',
      path: '/login',
      component: Login
    },
    {
      name: 'account',
      path: '/account',
      component: Account,
      beforeEnter: signedInCheck
    },
    {
      name: 'cart',
      path: '/cart',
      component: Cart,
      beforeEnter: signedInCheck
    },
    {
      name: 'about',
      path: '/about',
      component: About
    },
    {
      name: 'search',
      path: '/search/:query',
      component: ProductSearch
    },
    {
      name: 'catalog',
      path: '/catalog',
      component: ProductCatalog
    },
    {
      name: 'offers',
      path: '/offers',
      component: ProductOffers
    },
    {
      name: 'single-product',
      path: '/product/:id',
      component: ProductSingle
    },
    {
      name: 'view-order',
      path: '/order/:id',
      component: ViewOrder
      //beforeEnter: signedInCheck
    }
  ]
})

function signedInCheck(to, from, next) {
  const user = auth.user()
  if (!user || !user.username) {
    next('/login')
  } else {
    next()
  }
}

export default router
