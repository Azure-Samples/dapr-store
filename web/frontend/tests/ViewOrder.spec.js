import { describe, it, vi, expect } from 'vitest'
import { mount, flushPromises } from '@vue/test-utils'
import router from '@/router'

import ProductSearch from '@/views/ViewOrder.vue'

vi.mock('@/services/api')
vi.mock('@/services/auth')

describe('ViewOrder.vue', () => {
  it('shows order details', async () => {
    router.push('/order/ord-mock')
    await router.isReady()

    const wrapper = mount(ProductSearch, {
      global: {
        plugins: [router]
      }
    })

    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
