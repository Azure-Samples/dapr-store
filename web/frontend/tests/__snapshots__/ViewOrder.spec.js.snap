// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewOrder.vue > shows order details 1`] = `
"<div>
  <div class="d-flex justify-content-between mb-2">
    <h1>View Order Status</h1>
    <!--button class="btn btn-lg btn-success" @click="loadOrder()"><i class="fa-solid fa-rotate"></i> &nbsp; Refresh</button-->
  </div>
  <!--v-if-->
  <!--v-if-->
  <div data-v-f415f84a="" class="card order p-4">
    <div data-v-f415f84a="" class="d-flex justify-content-between mb-2">
      <h2 data-v-f415f84a="">Mock order</h2>
      <!--v-if-->
    </div>
    <h2 data-v-f415f84a=""> Status: <span data-v-f415f84a="" class="text-capitalize order-status order-new">new</span></h2>
    <ul data-v-f415f84a="">
      <li data-v-f415f84a="">Order Id: ord-mock</li>
      <li data-v-f415f84a="">Amount: £22.4</li>
      <li data-v-f415f84a="">Items:</li>
      <ul data-v-f415f84a="">
        <li data-v-f415f84a="">2 x Paisley Cravat Ascot Tie — £11.2</li>
      </ul>
    </ul>
  </div>
</div>"
`;
